spring-boot-admin-client/src/main/java/de/codecentric/boot/admin/client/registration/RegistrationApplicationListener.java

```diff
@@ -94,4 +95,26 @@ public void setAutoRegister(boolean autoRegister) {
     public void setRegisterPeriod(Duration registerPeriod) {
         this.registerPeriod = registerPeriod;
     }
+
+    @Override
+    public void afterPropertiesSet() {
+        taskScheduler = registrationTaskScheduler();
```

Comment:
I'd rather instantiate the TaskScheduler in the constructor and also the `afterPropertiesSet()` of the TaskScheduler must be invoked!